components/org.wso2.carbon.identity.conditional.auth.functions.choreo/src/main/java/org/wso2/carbon/identity/conditional/auth/functions/choreo/CallChoreoFunctionImpl.java

```diff
@@ -86,117 +121,24 @@ public void callChoreo(Map<String, String> connectionMetaData, Map<String, Objec
                 }
 
                 String tenantDomain = authenticationContext.getTenantDomain();
-
-                HttpPost request = new HttpPost(epUrl);
-                request.setHeader(ACCEPT, TYPE_APPLICATION_JSON);
-                request.setHeader(CONTENT_TYPE, TYPE_APPLICATION_JSON);
-
-                String apiKey;
-                if (StringUtils.isNotEmpty(connectionMetaData.get(API_KEY_VARIABLE_NAME))) {
-                    apiKey = connectionMetaData.get(API_KEY_VARIABLE_NAME);
+                AccessTokenRequestHelper accessTokenRequestHelper = new AccessTokenRequestHelper(
+                        connectionMetaData, asyncReturn, authenticationContext, payloadData);
+                String accessToken = choreoAccessTokenCache.getValueFromCache(accessTokenRequestHelper.getConsumerKey(),
+                        tenantDomain);
+                if (StringUtils.isNotEmpty(accessToken) && !isTokenExpired(accessToken)) {
+                    accessTokenRequestHelper.callChoreoEndpoint(accessToken);
                 } else {
-                    String apiKeyAlias = connectionMetaData.get(API_KEY_ALIAS_VARIABLE_NAME);
-                    apiKey = getResolvedSecret(apiKeyAlias);
-                }
-                request.setHeader(API_KEY, apiKey);
-
-                JSONObject jsonObject = new JSONObject();
-                for (Map.Entry<String, Object> dataElements : payloadData.entrySet()) {
-                    jsonObject.put(dataElements.getKey(), dataElements.getValue());
+                    LOG.debug("Requesting the access token from Choreo");
```

Comment:
Don't we need to check `LOG.isDebugEnabled()`?

Reply:
IMO, we don't need to check `LOG.isDebugEnabled()` since we don't have any string concatenation operations or such. 
Ref: https://stackoverflow.com/questions/6504407/is-there-a-need-to-do-a-iflog-isdebugenabled-check